app/lib/features/chat_ng/providers/notifiers/chat_room_messages_notifier.dart

```diff
@@ -84,6 +84,7 @@ ChatRoomState handleDiff(
   RoomMessageDiff diff, // the diff to apply
 ) {
   final action = diff.action();
+  // the diff is applied in reverse order as the animated list is provided to UI in reverse .i.e. for improved scrolling
```

Comment:
Tried various ways to fix scrolling without reversing the list but unfortunately, couldn't find a way to have optimal scroll offsets for various requirements we have for the chat .i.e. encountering offset bugs like not scrolling over all the bottom and how the pagination loading affects the UI refresh before. 

It seems like keeping `reverse:true` is only thing fixing the offset issue, which is also a normal behavior of the chat messages.

Reply:
The change-set only addresses the updates over indexing, as we have reversed the list.